packages/amplify_authenticator/lib/src/widgets/field_config.dart

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+*/
+
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+
+/// Configuration for defining content of radio and switch types in [AuthenticatorFormField].
+class AuthenticatorInputConfig {
+  final List<InputSelection> selections;
```

Comment:
This is implicitly `List<InputSelection<dynamic>>`. I think you want this to be `List<InputSelection<T>>`, and to make `AuthenticatorInputConfig` generic, `AuthenticatorInputConfig<T>`.

These is another location in the PR where inputconfig!.selections[0].value has to be cast to FieldValue. This is the reason that cast is needed.

Reply:
I changed AuthenticatorInputConfig to an interface which the mixins each implement, which forces each class that uses the mixin to implement the correct config.

Reply:
Also, the name is now `SelectableConfig`.